audio/src/audio.py

```diff
@@ -1,73 +1,161 @@
-import random
-import string
+import threading
 
 import os
+import time
+
 os.environ['PYGAME_HIDE_SUPPORT_PROMPT'] = "hide"
 import pygame as pg
 
 class AudioMixer:
 
-    def __init__(self, nof_max_sounds=20):
+    def __init__(self, event_callback, nof_max_sounds=20):
         pg.mixer.init()
         pg.init()
         pg.mixer.set_num_channels(nof_max_sounds)
+        self._event_callback = event_callback
         self._sounds = {}
+        self._sound_infos = {}
+        self._end_events = {}
+        self._current_end_event = 1000
+
+        self._event_handle_thread = threading.Thread(target=self._event_handler)
+        self._event_handle_thread.start()
 
-    def _generate_id(self):
+    def _event_handler(self):
```

Comment:
"event" is used in multiple places, and I'm having difficulty keeping track. This one handles events from pygame, right? Maybe call it _pygame_event_handler?